module/zfs/zil.c

```diff
@@ -723,9 +725,39 @@ zil_create(zilog_t *zilog)
 	 * (zh is part of the MOS, so we cannot modify it in open context.)
 	 */
 	if (tx != NULL) {
+		/*
+		 * If "zilsaxattr" feature is enabled on zpool, then activate
+		 * it.
```

Comment:
`then activate it now when we're creating the ZIL chain. We can't wait with this until we write the first xattr log record because we need to wait for the feature activation to sync out.`

Reply:
Current patch seems working. So, would discuss offline, to understand bit more on what change is necessary here.

Reply:
Updated comment as per discussion.